src/System.Security.Cryptography.X509Certificates/tests/Cert.cs

```diff
@@ -0,0 +1,159 @@
+        {
+            lock (SyncObject)
+            {
+                return new X509Certificate2(rawData, password, keyStorageFlags);
+            }
+        }
+
+        public static X509Certificate2 New(string fileName, string password, X509KeyStorageFlags keyStorageFlags)
+        {
+            lock (SyncObject)
+            {
+                return new X509Certificate2(fileName, password, keyStorageFlags);
+            }
+        }
+
+        //
+        // The Import() methods have an overload for each X509Certificate2Collection.Import() overload.
+        //
+
+        // Do not refactor this into a call to Import(byte[], string, X509KeyStorageFlags). The test meant to exercise
+        // the api that takes only one argument.
+        public static ImportedCollection Import(byte[] rawData)
+        {
+            X509Certificate2Collection collection = new X509Certificate2Collection();
+            lock (SyncObject)
+            {
+                collection.Import(rawData);
+            }
+            return new ImportedCollection(collection);
+        }
+
+        public static ImportedCollection Import(byte[] rawData, string password, X509KeyStorageFlags keyStorageFlags)
+        {
+            X509Certificate2Collection collection = new X509Certificate2Collection();
+            lock (SyncObject)
+            {
+                collection.Import(rawData, password, keyStorageFlags);
+            }
+            return new ImportedCollection(collection);
+        }
+
+        // Do not refactor this into a call to Import(string, string, X509KeyStorageFlags). The test meant to exercise
+        // the api that takes only one argument.
+        public static ImportedCollection Import(string fileName)
+        {
+            X509Certificate2Collection collection = new X509Certificate2Collection();
+            lock (SyncObject)
+            {
+                collection.Import(fileName);
+            }
+            return new ImportedCollection(collection);
+        }
+
+        public static ImportedCollection Import(string fileName, string password, X509KeyStorageFlags keyStorageFlags)
+        {
+            X509Certificate2Collection collection = new X509Certificate2Collection();
+            lock (SyncObject)
+            {
+                collection.Import(fileName, password, keyStorageFlags);
+            }
+            return new ImportedCollection(collection);
+        }
+
+        public static object SyncObject { get; } = new object();
+    }
+
+    //
+    // Wraps an X509Certificate2Collection in an IDisposable for easier cleanup.
+    //
+    internal sealed class ImportedCollection : IDisposable
+    {
+        public ImportedCollection(X509Certificate2Collection collection)
+        {
+            // Make an independent copy of the certs to dispose (in case the test mutates the collection after we return.)
+            _certs = new X509Certificate2[collection.Count];
```

Comment:
Making a collection dispose for this was on my mental todo list; but I hadn't considered at all the problem of the collection being modified during the test.  Good catch!